qupath-gui-fx/src/main/java/qupath/lib/gui/commands/PathObjectGridView.java

```diff
@@ -505,14 +496,12 @@ private void sortAndFilter() {
 						// pathclass is present and selected, or missing and we're showing unclassifier
 						&& (selectedClasses.contains(p.getPathClass()) || (p.getPathClass() == null && selectedClasses.contains(PathClass.NULL_CLASS)))
 		);
-		grid.getItems().setAll(filteredList);
+		Platform.runLater(() -> grid.getItems().setAll(filteredList));
```

Comment:
Show this always `runLater`, even if `Platform.isFxApplicationThread()` returns `true`?

Reply:
I figured it's unimportant as long as it is run later? I did add a check though if not